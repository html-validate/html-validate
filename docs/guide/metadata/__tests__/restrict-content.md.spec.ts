import HtmlValidate from "../../../../src/htmlvalidate";

const markup: { [key: string]: string } = {};
markup["tags"] = `<my-component>
  <button type="button">click me!</button>
</my-component>`;
markup["exclude"] = `<my-component>
  <div>allowed</div>
  <span>also allowed</span>
  <h1>not allowed</h1>
</my-component>`;
markup["descendants"] = `<my-component>
<!-- the div itself is allowed -->
  <div>
    <footer>
      sectioning element can no longer be used
    </footer>
    <my-component>
      nor can the component be nested
    </my-component>
  </div>
  <span>also allowed</span>
  <h1>not allowed</h1>
</my-component>`;

describe("docs/guide/metadata/restrict-content.md", () => {
	it("inline validation: tags", () => {
		const htmlvalidate = new HtmlValidate({"elements":["html5",{"my-component":{"flow":true,"permittedContent":["span","strong","em"]}}],"extends":["html-validate:recommended"]});
		const report = htmlvalidate.validateString(markup["tags"]);
		expect(report.results).toMatchSnapshot();
	});
	it("inline validation: exclude", () => {
		const htmlvalidate = new HtmlValidate({"elements":["html5",{"my-component":{"flow":true,"permittedContent":[{"exclude":"@heading"}]}}],"extends":["html-validate:recommended"]});
		const report = htmlvalidate.validateString(markup["exclude"]);
		expect(report.results).toMatchSnapshot();
	});
	it("inline validation: descendants", () => {
		const htmlvalidate = new HtmlValidate({"elements":["html5",{"footer":{"flow":true,"sectioning":true},"my-component":{"flow":true,"permittedDescendants":[{"exclude":["@sectioning","my-component"]}]}}],"extends":["html-validate:recommended"]});
		const report = htmlvalidate.validateString(markup["descendants"]);
		expect(report.results).toMatchSnapshot();
	});
});
