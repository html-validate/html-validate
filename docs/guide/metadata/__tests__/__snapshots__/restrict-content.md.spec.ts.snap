// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`docs/guide/metadata/restrict-content.md inline validation: descendants 1`] = `
[
  {
    "errorCount": 2,
    "filePath": "inline",
    "messages": [
      {
        "column": 6,
        "context": {
          "ancestor": "<my-component>",
          "child": "<footer>",
          "kind": "descendant",
        },
        "line": 4,
        "message": "<footer> element is not permitted as a descendant of <my-component>",
        "offset": 63,
        "ruleId": "element-permitted-content",
        "ruleUrl": "https://html-validate.org/rules/element-permitted-content.html",
        "selector": "my-component > div > footer",
        "severity": 2,
        "size": 6,
      },
      {
        "column": 6,
        "context": {
          "ancestor": "<my-component>",
          "child": "<my-component>",
          "kind": "descendant",
        },
        "line": 7,
        "message": "<my-component> element is not permitted as a descendant of <my-component>",
        "offset": 137,
        "ruleId": "element-permitted-content",
        "ruleUrl": "https://html-validate.org/rules/element-permitted-content.html",
        "selector": "my-component > div > my-component",
        "severity": 2,
        "size": 12,
      },
    ],
    "source": "<my-component>
<!-- the div itself is allowed -->
  <div>
    <footer>
      sectioning element can no longer be used
    </footer>
    <my-component>
      nor can the component be nested
    </my-component>
  </div>
  <span>also allowed</span>
  <h1>not allowed</h1>
</my-component>",
    "warningCount": 0,
  },
]
`;

exports[`docs/guide/metadata/restrict-content.md inline validation: exclude 1`] = `
[
  {
    "errorCount": 1,
    "filePath": "inline",
    "messages": [
      {
        "column": 4,
        "context": {
          "child": "<h1>",
          "kind": "content",
          "parent": "<my-component>",
        },
        "line": 4,
        "message": "<h1> element is not permitted as content under <my-component>",
        "offset": 67,
        "ruleId": "element-permitted-content",
        "ruleUrl": "https://html-validate.org/rules/element-permitted-content.html",
        "selector": "my-component > h1",
        "severity": 2,
        "size": 2,
      },
    ],
    "source": "<my-component>
  <div>allowed</div>
  <span>also allowed</span>
  <h1>not allowed</h1>
</my-component>",
    "warningCount": 0,
  },
]
`;

exports[`docs/guide/metadata/restrict-content.md inline validation: tags 1`] = `
[
  {
    "errorCount": 1,
    "filePath": "inline",
    "messages": [
      {
        "column": 4,
        "context": {
          "child": "<button>",
          "kind": "content",
          "parent": "<my-component>",
        },
        "line": 2,
        "message": "<button> element is not permitted as content under <my-component>",
        "offset": 18,
        "ruleId": "element-permitted-content",
        "ruleUrl": "https://html-validate.org/rules/element-permitted-content.html",
        "selector": "my-component > button",
        "severity": 2,
        "size": 6,
      },
    ],
    "source": "<my-component>
  <button type="button">click me!</button>
</my-component>",
    "warningCount": 0,
  },
]
`;
